### PR TITLE
JDBC client timeouts on long requests

### PR DESCRIPTION
**Pull request by BentsiLeviav**
Document the required configuration to resolve the issue.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 363f417d2876ee522a04bffb3e3047644969962d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/1535/363f417d2876ee522a04bffb3e3047644969962d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tr><td>Vercel</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/en/integrations/language-clients/java/index.md [hunk 1]
```diff
@@ -378,6 +378,58 @@ try (PreparedStatement stmt = conn.prepareStatement(
...
+   - `net.ipv4.tcp_keepalive_probes`: 9
+   - `net.ipv4.tcp_keepalive_time`: 60 (You may consider lowering this value from the default 300 seconds)
+
+2. After modifying the kernel parameters, apply the changes by running the following command:
+
+   ```shell
+   sudo sysctl -p
+   ```
+
+After Setting those settings, you need to ensure that your client enables the Keep Alive option on the socket. Currently, you must use Apache HTTP Client library, as the other two HTTP client libraries in clickhouse-java do not allow setting socket options:
```
Comment (mshustov) [inline docs/en/integrations/language-clients/java/index.md:424]: Optional: Can we move to a separate section with a short guide on changing the default HTTP library?

**Review by mshustov — APPROVED** [review]

File: docs/en/integrations/language-clients/java/index.md (hunk 1 above)
Comment (BentsiLeviav) [reply docs/en/integrations/language-clients/java/index.md]: Added. Could you please review it as well?

File: docs/en/integrations/language-clients/java/index.md [hunk 2]
```diff
@@ -378,6 +378,86 @@ try (PreparedStatement stmt = conn.prepareStatement(
     Assert.assertFalse(rs.next());
 }
 ```
+<br/>
+
+#### Configuring HTTP library
+
+The ClickHouse JDBC connector supports three HTTP libraries: HttpClient, HttpURLConnection, and Apache HttpClient.
```
Comment (mshustov) [inline docs/en/integrations/language-clients/java/index.md:385]:
```suggestion
The ClickHouse JDBC connector supports three HTTP libraries: [HttpClient](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.html), [HttpURLConnection](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/HttpURLConnection.html), and [Apache HttpClient](https://hc.apache.org/httpcomponents-client-5.2.x/).
```

File: docs/en/integrations/language-clients/java/index.md [hunk 3]
```diff
@@ -378,6 +378,86 @@ try (PreparedStatement stmt = conn.prepareStatement(
...
+
+:::note
+HttpClient is only supported in JDK 11 or above.
+:::
+
+You can change the default HTTP library used by the ClickHouse JDBC connector, by adding the following property:
```
Comment (mshustov) [inline docs/en/integrations/language-clients/java/index.md:391]:
```suggestion
By default, the JDBC driver uses `HttpClient`. You can change the default HTTP library used by the ClickHouse JDBC connector, by adding the following property:
```

**Review by mshustov — APPROVED** [review]

**Review by mshustov — APPROVED** [review]

**Pull request merged**